### PR TITLE
{base}[system/] bcl-convert v4.0.3

### PR DESCRIPTION
**Pull request by garadar**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/b/bcl-convert/bcl-convert-4.0.3.eb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+toolchain = {'name': 'system', 'version': ''}
+
+sources = ["https://webdata.illumina.com/downloads/software/bcl-convert/bcl-convert-4.0.3-2.el7.x86_64.rpm"]
+checksums = ["be4f3d1d31cb36618f909cdbc0ed0fe70ef57a645e46b421b6e1a7f90c6c9430"]
+
+sanity_check_commands = ["%(name)s --version"]
+sanity_check_paths = {
+    'files': ["usr/bin/bcl-convert"],
+    'dirs': []
+}
```
Comment (verdurin) [inline easybuild/easyconfigs/b/bcl-convert/bcl-convert-4.0.3.eb:19]: Please add the missing `moduleclass`:

```suggestion
}

moduleclass = 'bio'
```

**Review by verdurin — CHANGES_REQUESTED** [review]

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/22c82a8f3cb73ae3d92764f97b1f1600 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/83b8a3e0011ad51786f260a5540fa76d for a full test report.

**Review by verdurin — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/b/bcl-convert/bcl-convert-4.0.3.eb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+name = 'bcl-convert'
+version = '4.0.3'
+homepage = 'https://www.illumina.com/products/by-type/informatics-products/basespace-sequence-hub/apps/bcl-convert.html'
+description = """The BCL Convert App generates demultiplexed FASTQ files from a run as input. The core algorithm is BCL Convert.
+
+for more information: https://www.illumina.com/products/by-type/informatics-products/basespace-sequence-hub/apps/bcl-convert.html
+"""
+toolchain = {'name': 'system', 'version': ''}
+
+sources = ["https://webdata.illumina.com/downloads/software/%(name)s/%(name)s-%(version)s.el7.x86_64.rpm"]
```
Comment (lexming) [inline easybuild/easyconfigs/b/bcl-convert/bcl-convert-4.0.3.eb:12]: I could not download the RPM, is the download restricted? If so, please add a comment that registration/license is needed

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
centos7.lan - Linux CentOS Linux 7.9.2009, x86_64, Intel Core Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/8a59d988e315c9e4c5767c71f64e50a9 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
centos7.lan - Linux CentOS Linux 7.9.2009, x86_64, Intel Core Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/3fbfd459f97033f92f6054aae1b0c6a3 for a full test report.

File: easybuild/easyconfigs/b/bcl-convert/bcl-convert-4.0.3.eb (hunk 2 above)
Comment (verdurin) [reply easybuild/easyconfigs/b/bcl-convert/bcl-convert-4.0.3.eb:14]: @lexming I didn't see any problem downloading the RPM. I did, however, need to install `rpmrebuild` on the build node.

Comment (lexming) [reply easybuild/easyconfigs/b/bcl-convert/bcl-convert-4.0.3.eb:14]: @verdurin https://webdata.illumina.com/downloads/software/bcl-convert/bcl-convert-4.0.3.el7.x86_64.rpm is throws a 404 for me. On the other hand, `rpmbuild` should not be needed to install any rpm package.

**Comment by verdurin** [comment]
On Mon, Nov 21, 2022 at 02:52:15AM -0800, Alex Domingo wrote:
> @lexming commented on this pull request.
> 
> 
> 
> > @@ -0,0 +1,21 @@
> +easyblock = 'Rpm'
> +
> +name = 'bcl-convert'
> +version = '4.0.3'
> +homepage = 'https://www.illumina.com/products/by-type/informatics-products/basespace-sequence-hub/apps/bcl-convert.html'
> +description = """The BCL Convert App generates demultiplexed FASTQ files from a run as input. The core algorithm is BCL Convert.
> +
> +for more information: https://www.illumina.com/products/by-type/informatics-products/basespace-sequence-hub/apps/bcl-convert.html
> +"""
> +toolchain = {'name': 'system', 'version': ''}
> +
> +sources = ["https://webdata.illumina.com/downloads/software/%(name)s/%(name)s-%(version)s.el7.x86_64.rpm"]
> 
> @verdurin https://webdata.illumina.com/downloads/software/bcl-convert/bcl-convert-4.0.3.el7.x86_64.rpm is throws a 404 for me. On the other hand, `rpmbuild` should not be needed to install any rpm package.
> 

There's been a minor release - it's now 4.0.3-2 - and that downloads for
me.

That said, now I see there's a duplicate from @boegel:
https://github.com/easybuilders/easybuild-easyconfigs/pull/16351

**Comment by lexming** [comment]
@verdurin thanks for the clarification. Since the dowload does not work anymore, I suggest to close this PR in favour of https://github.com/easybuilders/easybuild-easyconfigs/pull/16351

**Comment by verdurin** [comment]
On Mon, Nov 21, 2022 at 04:24:15AM -0800, Alex Domingo wrote:
> @verdurin thanks for the clarification. Since the dowload does not work anymore, I suggest to close this PR in favour of https://github.com/easybuilders/easybuild-easyconfigs/pull/16351
> 

The download does work though - the version has already been updated...
(you may have missed the commit that did so)

**Comment by lexming** [comment]
@verdurin indeed, I missed the last commit. Nonetheless, previous PR #16531 is now merged, so I'll close this PR.
@garadar thanks for your PR, this easyconfig has been already merged in #16531

**Pull request closed without merging**